llumnix/backends/utils.py

```diff
@@ -53,34 +96,30 @@ async def put_nowait_to_servers(self,
             # The lifetime of AsyncPutQueueActor is the same as the lifetime of the instance actor,
             # so we do not handling exception here.
             self.engine_actor_handle = ray.get_actor(get_instance_name(self.instance_id), namespace="llumnix")
-        tasks = []
-        for server_id, req_outputs in server_request_outputs.items():
-            server_info = server_info_dict[server_id]
-            set_timestamp(req_outputs, 'engine_actor_put_queue_timestamp', time.time())
-            tasks.append(asyncio.create_task(self.request_output_queue_client.put_nowait(req_outputs, server_info)))
-        rets = await asyncio.gather(*tasks, return_exceptions=True)
-        request_ids = []
-        for idx, ret in enumerate(rets):
-            # If exception occurs during sending message from queue client to queue server,
-            # AsyncPutQueueActor will not die.
-            if isinstance(ret, Exception):
-                server_id = list(server_request_outputs.keys())[idx]
-                server_info = server_info_dict[server_id]
-                logger.error("Server {} is dead, exception: {}.".format(server_id, ret))
-                if self.request_output_queue_type == QueueType.ZMQ:
-                    logger.debug("request output queue ip: {}, port: {}".format(server_info.request_output_queue_ip,
-                                                                                server_info.request_output_queue_port))
-                req_outputs = list(server_request_outputs.values())[idx]
-                request_ids = []
-                for req_output in req_outputs:
-                    assert isinstance(req_output, LlumnixRequestOuput)
-                    request_ids.append(req_output.request_id)
-                await asyncio_wait_for_with_timeout(self.engine_actor_handle.abort.remote(request_ids))
+        aborted_request_ids = await put_nowait_to_servers_func(
+            self.request_output_queue_type, self.request_output_queue_client, server_request_outputs, server_info_dict
+        )
+        if aborted_request_ids:
+            await asyncio_wait_for_with_timeout(self.engine_actor_handle.abort.remote(aborted_request_ids))
 
     def stop(self):
         if self.request_output_queue_type == QueueType.ZMQ:
             self.request_output_queue_client.close()
 
+async def put_nowait_to_servers(
```

Comment:
maybe use different names for the two put_nowait_to_servers (actor and w/o actor) to differentiate them